okhttp-clients/src/main/java/com/palantir/conjure/java/okhttp/RemotingOkHttpCall.java

```diff
@@ -342,14 +361,19 @@ private boolean shouldRetry(IOException exception, Optional<Duration> backoff) {
     private void scheduleExecution(
             Duration backoff,
             Tags.AttemptSpan attemptSpan,
-            Runnable execution) {
+            Consumer<Callback> enqueueFunction,
+            Callback callback) {
         DetachedSpan backoffSpan = attemptSpan.attemptSpan().childDetachedSpan("OkHttp: backoff-with-jitter");
 
         // TODO(rfink): Investigate whether ignoring the ScheduledFuture is safe, #629.
         schedulingExecutor.schedule(
-                () -> executionExecutor.submit(() -> {
+                () -> executionExecutor.execute(() -> {
                     backoffSpan.complete();
-                    execution.run();
+                    if (isCanceled()) {
+                        callback.onFailure(this, new SafeIoException("Request is cancelled"));
+                    } else {
+                        enqueueFunction.accept(callback);
+                    }
```

Comment:
If the calling thread is interrupted while waiting to retry, we don't want to send more requests.

Reply:
Alternatively we could track futures, but that gets gnarly and easy to break.